components/ppp_link/ppp_link.c

```diff
@@ -10,29 +10,31 @@
 #include "esp_netif_ppp.h"
 #include "netif/ppp/ppp.h"
 #include "ppp_link.h"
+#include <sys/param.h>
+
+#define PPP_LINK_TASK_STACK_SIZE (3 * 1024)
+#define PPP_LINK_TASK_PRIO 100
+
+#define MAX_PPP_FRAME_SIZE (PPP_MAXMRU + 10) // 10 bytes of ppp framing around max 1500 bytes information
 
-#define MIN(x, y) (((x) < (y)) ? (x) : (y))
 static const char *TAG = "ppp_link";
-static QueueHandle_t uart_event_queue = NULL;
+static QueueHandle_t uart_event_queue;
 static int current_phase = PPP_PHASE_DEAD;
 static ppp_link_config_t config;
-#define MAX_PPP_FRAME_SIZE (PPP_MAXMRU + 10) // 10 bytes of ppp framing around max 1500 bytes information
-
-esp_err_t esp_netif_start(esp_netif_t *esp_netif);
 
 static void on_ppp_changed(void *arg, esp_event_base_t event_base,
                            int32_t event_id, void *event_data) {
-  if (event_id >= NETIF_PP_PHASE_OFFSET) {
-    current_phase = event_id - NETIF_PP_PHASE_OFFSET;
-  }
+    if (event_id >= NETIF_PP_PHASE_OFFSET) {
+        current_phase = event_id - NETIF_PP_PHASE_OFFSET;
+    }
 }
 
 static esp_err_t on_ppp_transmit(void *h, void *buffer, size_t len) {
     size_t free_size = 0;
     ESP_ERROR_CHECK(uart_get_tx_buffer_free_size(config.uart, &free_size));
 
     if (unlikely(free_size < len)) {
-        //ESP_LOGV(TAG, "Uart tx buffer full. free_size: %d len: %d", free_size, len);
+        ESP_LOGW(TAG, "Uart TX buffer full. free_size: %d len: %d", free_size, len);
```

Comment:
Logging this will kill the performance of the link if testing with iperf, iperf will fill up the buffer by design, and keep it full.

Reply:
fixed